route53.tf

```diff
@@ -1,15 +1,17 @@
 data "aws_route53_zone" "selected" {
   count = length(var.alias)
 
-  name = lookup(var.alias[count.index], "domain", "")
+  name         = var.alias[count.index].domain
+  vpc_id       = lookup(var.alias[count.index], "private_zone", false) ? data.aws_vpc.selected.id : null
+  private_zone = lookup(var.alias[count.index], "private_zone", false)
 }
 
 resource "aws_route53_record" "default" {
   count = length(var.alias)
 
   zone_id = element(data.aws_route53_zone.selected.*.zone_id, count.index)
-  name    = lookup(var.alias[count.index], "hostname", var.name)
+  name    = var.alias[count.index].hostname
```

Comment:
Please revert this change, otherwise update the documentation to indicate that `var.name` is not the default.